packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -0,0 +1,657 @@
+  /// Unicode "THUMBS UP SIGN + MEDIUM SKIN TONE MODIFIER", "👍🏽", should be
+  /// counted as a single character, but because it is a combination of two
+  /// Unicode scalar values, '\u{1F44D}\u{1F3FD}', it is counted as two
+  /// characters.
+  ///
+  /// See also:
+  ///
+  ///  * [LengthLimitingTextInputFormatter] for more information on how it
+  ///    counts characters, and how it may differ from the intuitive meaning.
+  final int maxLength;
+
+  /// If true, prevents the field from allowing more than [maxLength]
+  /// characters.
+  ///
+  /// If [maxLength] is set, [maxLengthEnforced] indicates whether or not to
+  /// enforce the limit, or merely provide a character counter and warning when
+  /// [maxLength] is exceeded.
+  final bool maxLengthEnforced;
+
+  /// {@macro flutter.widgets.editableText.onChanged}
+  final ValueChanged<String> onChanged;
+
+  /// {@macro flutter.widgets.editableText.onEditingComplete}
+  final VoidCallback onEditingComplete;
+
+  /// {@macro flutter.widgets.editableText.onSubmitted}
+  final ValueChanged<String> onSubmitted;
+
+  /// {@macro flutter.widgets.editableText.inputFormatters}
+  final List<TextInputFormatter> inputFormatters;
+
+  /// Disables the text field when false.
+  ///
+  /// Text fields in disabled states have a light grey background and don't
+  /// respond to touch events including the [prefix], [suffix] and the clear
+  /// button.
+  final bool enabled;
+
+  /// {@macro flutter.widgets.editableText.cursorWidth}
+  final double cursorWidth;
+
+  /// {@macro flutter.widgets.editableText.cursorRadius}
+  final Radius cursorRadius;
+
+  /// The color to use when painting the cursor.
+  ///
+  /// Defaults to the standard iOS blue color. Cannot be null.
+  final Color cursorColor;
+
+  /// The appearance of the keyboard.
+  ///
+  /// This setting is only honored on iOS devices.
+  ///
+  /// If null, defaults to [Brightness.light].
+  final Brightness keyboardAppearance;
+
+  /// {@macro flutter.widgets.editableText.scrollPadding}
+  final EdgeInsets scrollPadding;
+
+  @override
+  _CupertinoTextFieldState createState() => _CupertinoTextFieldState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+
+    properties.add(DiagnosticsProperty<TextEditingController>('controller', controller, defaultValue: null));
+    properties.add(DiagnosticsProperty<FocusNode>('focusNode', focusNode, defaultValue: null));
+    properties.add(DiagnosticsProperty<BoxDecoration>('decoration', decoration));
+    properties.add(DiagnosticsProperty<EdgeInsetsGeometry>('padding', padding));
+    properties.add(StringProperty('placeholder', placeholder));
+    if (prefix != null) {
+      properties.add(DiagnosticsProperty<OverlayVisibilityMode>('prefix', prefixMode));
+    }
+    if (suffix != null) {
+      properties.add(DiagnosticsProperty<OverlayVisibilityMode>('suffix', suffixMode));
+    }
```

Comment:
we should always add these, just add them with defaultValue: null

Reply:
done